cmd/kubelet/app/server.go

```diff
@@ -297,8 +297,7 @@ func initConfigz(kc *componentconfig.KubeletConfiguration) (*configz.Config, err
 }
 
 func run(s *options.KubeletServer, kubeDeps *kubelet.KubeletDeps) (err error) {
-	// TODO: this should be replaced by a --standalone flag
-	standaloneMode := (len(s.APIServerList) == 0 && !s.RequireKubeConfig)
+	standaloneMode := s.Standalone
```

Comment:
The issue would be that the new flag would be false by default, but people would expect standalone mode to be on given an empty APIServerList. For now we can probably handle that by saying `(len(s.APIServerList) == 0 || s.Standalone) && !s.RequireKubeConfig`, and adding a todo to move to `s.Standalone && !s.RequireKubeConfig` when `APIServerList` is finally removed.

I might have been wrong about there being an issue in the `RequireKubeConfig` case.   @smarterclayton @errordeveloper is standalone mode even possible with `RequireKubeConfig`? (Given the title of https://github.com/kubernetes/kubernetes/commit/fe768dd405691ea2052c0607b447d35ccce4b71e "Do not go into standalone mode when `--require-kubeconfig` is passed").